src/test/app/ReducedOffer_test.cpp

```diff
@@ -606,13 +609,193 @@ class ReducedOffer_test : public beast::unit_test::suite
+                    // this to verify that the largest allowable TakerGets
+                    // was computed.
+                    if (badRate == 0)
+                    {
+                        STAmount const tweakedTakerGets(
+                            aliceReducedOffer.in.issue(),
+                            aliceReducedOffer.in.mantissa() + 1,
+                            aliceReducedOffer.in.exponent(),
+                            aliceReducedOffer.in.negative());
+                        STAmount const tweakedRate =
+                            Quality(
+                                Amounts{aliceReducedOffer.in, tweakedTakerGets})
+                                .rate();
+                        BEAST_EXPECT(tweakedRate > initialRate);
+                    }
+#if 0
+                    std::cout << "Placed rate: " << initialRate
+                              << "; in-ledger rate: " << inLedgerRate
+                              << "; TakerPays: " << aliceReducedOffer.in
+                              << "; TakerGets: " << aliceReducedOffer.out
+                              << std::endl;
+// #else
+                    std::string_view filler = badRate ? "**" : "  ";
+                    std::cout << "| " << aliceReducedOffer.in << "` | `"
+                              << aliceReducedOffer.out << "` | `" << initialRate
+                              << "` | " << filler << "`" << inLedgerRate << "`"
+                              << filler << std::endl;
+#endif
+                }
+
+                // In preparation for the next iteration make sure all three
+                // offers are gone from the ledger.
+                cleanupOldOffers(
+                    env,
+                    {{alice, aliceOfferSeq},
+                     {bob, bobOfferSeq},
+                     {carol, carolOfferSeq}});
+                return badRate;
+            };
+
+            constexpr int loopCount = 100;
+            unsigned int blockedCount = 0;
+            {
+                STAmount increaseGets = USD(0);
+                STAmount const step(increaseGets.issue(), 1, -8);
+                for (unsigned int i = 0; i < loopCount; ++i)
+                {
+                    blockedCount += exerciseOfferTrio(
+                        Amounts(drops(1642020), USD(1) + increaseGets));
+                    increaseGets += step;
+                }
+            }
+
+            // If fixReducedOffersV2 is enabled, then none of the test cases
+            // should produce a potentially blocking rate.
+            //
+            // Also verify that if fixReducedOffersV2 is not enabled then
+            // some of the test cases produced a potentially blocking rate.
+            if (features[fixReducedOffersV2])
+            {
+                BEAST_EXPECT(blockedCount == 0);
+            }
+            else
+            {
+                BEAST_EXPECT(blockedCount > 80);
```

Comment:
Shouldn't we check for the exact count of the blocking rates since the same tests are executed on every run? If this count changes in the next release then doesn't it indicate some change that needs to be investigated?

Reply:
Yeah, it's fine.